File: services/ingestion/telegram-scraper/models/RawTelegramEvent.py
from pydantic import BaseModel, Field
from shared.RawOsintEvent import RawOsintEvent


class TelegramEventData(BaseModel):
    event_type: str
    chat_id: int
    channel_title: str
    channel_main_lang: str
    msg_id: int


class RawTelegramEvent(RawOsintEvent):
    source: str = Field(default="telegram", init=False)
    data: TelegramEventData

File: services/processing/models/RawTelegramEvent.go
package models

import (
	"encoding/json"
	"fmt"
)

type RawTelegramEvent struct {
	rawOsintEvent
	Text            string `json:"text"`
	EventType       string `json:"event_type"`
	ChatID          int64  `json:"chat_id"`
	ChannelTitle    string `json:"channel_title"`
	ChannelMainLang string `json:"channel_main_lang"`
	MessageID       int    `json:"message_id"`
}

func (e *RawTelegramEvent) Unmarshal(data []byte) error {
	if err := json.Unmarshal(data, e); err != nil {
		return fmt.Errorf("RawTelegramEvent unmarshal error: %w", err)
	}
	return nil
}
